Generators

achess opened the pull request:
Created generators for job types:
- Synchronous
- Container aware
- Throttled
- Loner

Based on the [Controller generator on the Symfony 2.2 branch](https://github.com/sensio/SensioGeneratorBundle/blob/2.2/Generator/ControllerGenerator.php)

Note: The generators in master now extend from the [Generator](https://github.com/sensio/SensioGeneratorBundle/blob/master/Generator/Generator.php) class, which will allow to remove some code from the JobGenerator class when upgrading versions.

The generator allows to send all the information in one command:

```
php app/console resque:generate:job --job=AcmeDemoBundle:SendEmail --job-type=loner --no-interaction
```

Or run `php app/console resque:generate:job` and fill the information interactively:

![image](https://f.cloud.github.com/assets/4226583/417542/17c202ac-ac6e-11e2-88fa-6ad5b5e000b3.png)

Commit:
Create templates for jobs - closes #4

Commit:
Create command and generator to create jobs

Commit:
Fix cached exception on no interaction command

ShonM commented:
Looking good!

One thing before I merge this in - can you add a "default" option for job types that's none of the above? 

I'm thinking also that loner and throttled should take arguments for their appropriate fields, but with the work on Sonido moving quickly that will likely fall in at a later date, so ignore this sentence :P

Commit:
Add "default" and "none" job types

Commit:
Improve readability of job types

The input is not case sentitive, so let's write the instructions for the
job types CamelCased for easier reading.

achess commented:
- Added two new job types:
  - `Default`: implements JobInterface (and also set as the default job type)
  - `None`: Doesn't extend or implement anything.

Job types are not case sensitive, so, made the job types CamelCased on the command instructions for easier reading (ex. `ContainerAware`, `Throttled`)

ShonM commented:
Awesome! This all looks great, thank you :)
